### ISSUE TITLE
train epoch acc have no change

### ISSUE THREAD

**Issue by myegdfz** [Goldesel23/Siamese-Networks-for-One-Shot-Learning#16]
firstly thx for yr work, and since the version, i made some change in your code, but it works nothing, and every epoch the accurcy rate was not change, like this.

![4(`}32%7( `MGBF_$ZW`E%1](https://user-images.githubusercontent.com/48350357/110952270-fb4b4a00-8380-11eb-860f-7e9de156e845.png)

**Comment by tensorfreitas** [comment]
Hi! could you elaborate on what did you change to the code?

**Comment by myegdfz** [comment]
since the original not at version tf 2.4, so I change the _write_logs_to_tensorboard to adapt version 2.4, this one 
![image](https://user-images.githubusercontent.com/48350357/111135710-f3291f80-85b7-11eb-8d44-c86d39b67c5d.png)
, and  the  Modified_SGD class's learning_rate, because **lr** is the father class's attribute. 
![image](https://user-images.githubusercontent.com/48350357/111135935-2e2b5300-85b8-11eb-993c-7c9447661a76.png)

**Comment by myegdfz** [comment]
and i also have a question, i modified the structure of the net, replace the original convolutional layer with vgg16 and add a netvald layer before the last layer
![image](https://user-images.githubusercontent.com/48350357/111139051-9d567680-85bb-11eb-9a1e-157d35582095.png)
but loss will increase to 36.xx from 5.0. and I can't understand.

**Comment by tensorfreitas** [comment]
Thanks for the details. I would suggest for making sure that the original architecture will decrease the training error in a small dataset. 

Only after that I would start making adjustments to the model architecture and follow the same process. Make sure you test adam/modified SGD with different learning rates. By making big adjustments it's hard to identify the reason of non convergence. Hope I was helpful

**Comment by myegdfz** [comment]
do you mean if my initial loss is too high, it would be stoped very fast? maybe at iteration 4.

**Comment by tensorfreitas** [comment]
No, I was meaning that making sure you can guarantee first that the architecture converges in a small dataset and that the training loss is decreasing. 

I would suggest reading this blog post by Andrej Karpathy, especially the overfit section:

http://karpathy.github.io/2019/04/25/recipe/
